Service/SamlDataHydrateService.php

```diff
@@ -28,24 +28,36 @@ public function __construct(
      */
     public function hydrate(User $user, SamlDTO $samlDto): void
     {
-        $orgUnitId = $samlDto->getMemberOrganizationUnitId();
-        $orgUnitName = $samlDto->getMemberOrganizationUnitName();
+        $teamLead = $this->hydrateTeamLead($samlDto);
 
-        $memberTeam = $this->hydrateTeam($orgUnitName, $orgUnitId, $samlDto);
+        // Setup team user is member of
+        $memberUnitId = $samlDto->getMemberOrganizationUnitId();
+        $memberUnitName = $samlDto->getMemberOrganizationUnitName();
+        $memberTeam = $this->hydrateTeam($memberUnitName, $memberUnitId, $samlDto->managerEmail, $samlDto->managerName, $samlDto);
 
-        if ($samlDto->isTeamLead()) {
-            $teamLead = $user;
+        // Setup manager (personaleleder) as team lead for the team
+        $this->hydrateTeamLeadForTeam($memberTeam, $teamLead);
+
+        // Set user values and team membership
+        $this->hydrateUser($user, $samlDto, $teamLead, $memberTeam);
 
-            $teamLeadTeamId = $samlDto->getTeamLeadOrganizationUnitId();
-            $teamLeadTeamName = $samlDto->getTeamLeadOrganizationUnitName();
-            $teamLeadTeam = $this->hydrateTeam($teamLeadTeamName, $teamLeadTeamId, $samlDto);
+        if ($samlDto->isTeamLead()) {
+            $teamUnitId = $samlDto->getTeamLeadOrganizationUnitId();
+            $teamUnitName = $samlDto->getTeamLeadOrganizationUnitName();
 
+            $teamLeadTeam = $this->hydrateTeam($teamUnitName, $teamUnitId, $samlDto->emailAddress, $samlDto->displayName, $samlDto);
             $this->hydrateTeamLeadForTeam($teamLeadTeam, $user);
-        } else {
-            $teamLead = $this->hydrateTeamLead($memberTeam, $samlDto);
-        }
 
-        $this->hydrateUser($user, $samlDto, $teamLead, $memberTeam);
+            foreach ($user->getTeams() as $team) {
+                if ($team->isTeamlead($user)) {
+                    foreach ($team->getUsers() as $teamMember) {
+                        if (!in_array($teamMember->getUsername(), $samlDto->employeeList) && $teamMember->getUsername() !== $user->getUsername()) {
```

Comment:
Under the assumption that comparing strings are faster than searching an array, a (very small) performance gain might be achieved by switching the expressions around:

```suggestion
                        if ($teamMember->getUsername() !== $user->getUsername() && !in_array($teamMember->getUsername(), $samlDto->employeeList)) {
```

Reply:
Given that there can only be one team lead for each team (our business rules, not Kimai's) we will always have to do the array search N-1 times regardless of the order of expressions. Have refactored to use `isset()`